hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -431,27 +432,36 @@ public Vertex onSlidingWindowAggregate(SlidingWindowAggregatePhysicalRel rel) {
                             windowPolicy,
                             0,
                             aggregateOperation,
-                            resultMapping));
+                            resultMapping,
+                            watermarkedFieldsKeys.isEmpty()
+                                    ? watermarkKeysAssigner.getInputWatermarkKey(rel)
+                                    : watermarkedFieldsKeys.get(rel.watermarkedFields().findFirst()).getValue()));
```

Comment:
(this question may have more sense here)

is it enough to use first input watermark? what if the input has more watermarks (eg. it is a results of s2s join)?

Reply:
I guess, yes. `SlidingWindowP` works only with one watermark. If you assign a new watermark here, it just skip all input events.